src/main/java/Aly/Aly.java

```diff
@@ -0,0 +1,451 @@
+                    break;
+                case "mark":
+                case "unmark":
+                case "delete":
+                    index = handleTasks(firstWord, splitInput, index);
+                    writeTasksToFile(SAVE_FILE.toString());
+                    break;
+                case "exit":
+                    isExit = true;
+                    System.out.println(RETURNING_TO_MAIN_MENU_MESSAGE);
+                    break;
+                default:
+                    throw new InputMismatchException();
+                }
+            } catch (InputMismatchException e) {
+                System.out.println("Can follow instructions properly anot? Read carefully and try again!");
+            } catch (ArrayIndexOutOfBoundsException e) {
+                System.out.println("You got too much work already bro, task list full liao!");
+            } catch (IllegalFormatException e) {
+                System.out.println("Can wake up your idea anot, how to do anything without proper input???");
+            } catch (Exception e) {
+                System.out.println("An unexpected error occurred: " + e.getMessage());
+            } finally {
+                if (!isExit) {
+                    printLine();
+                }
+            }
+        }
+    }
+
+    private static void createFile(File file) {
+        System.out.println("Searching for " + SAVE_FILE + "...");
+        try {
+            if (file.createNewFile()) {
+                System.out.println("Cannot find file so I help you create already! File name: " + SAVE_FILE);
+            } else {
+                System.out.println("File already exists, I will edit that!");
+            }
+        } catch (IOException e) {
+            System.out.println("Error! File sucks because: " + e.getMessage());
+        }
+    }
+
+    private static void createDirectory() {
+        Path directory = SAVE_FILE.getParent(); // Get the parent directory
+        System.out.println("Searching for directory: " + directory);
+        if (directory != null && !Files.exists(directory)) {
+            try {
+                Files.createDirectories(directory);
+            } catch (IOException e) {
+                System.out.println("Idk why cannot make directory siah: " + e.getMessage());
+            }
+            System.out.println("Directory didn't exist but I made it for u liao! You're welcome!");
+        } else {
+            System.out.println("Directory already exists, I will check in there!");
+        }
+    }
+
+    //Reusable to print any list of tasks
+    private static void listTasks() {
+        printLine();
+        int count = 1;
+        System.out.println("Your task list:");
+        try {
+            for (Task listItem : taskList) {
+                if (listItem == null) {
+                    break;
+                }
+                System.out.println(count + "." + listItem);
+                count++;
+            }
+        } catch (Exception e) {
```

Comment:
Try to catch specific errors so that you could provide better print statements to guide the user!